warehouse/schema.go

```diff
@@ -403,3 +413,62 @@ func mergeUploadAndLocalSchemas(uploadSchema, schemaInWarehousePreUpload warehou
 	}
 	return mergedSchema
 }
+
+/*
+ * Given maximum table column count limit
+ * 1. Returns excess columns from each table in upload schema
+ * 2. For a new event, rudder reserved columns are not included in excluded schema
+ * 3. For an existing event, rudder reserved columns do not hold any priority
+ */
+func GetExcludedSchema(uploadSchema, schemaInWarehouse warehouseutils.SchemaT, maxColumnCount int) warehouseutils.SchemaT {
+	excludedSchema := warehouseutils.SchemaT{}
+	for tableName, columnMap := range uploadSchema {
+		// To avoid computational cost, comparing each schema to half of maxColumnCount. Most cases fall under this category.
+		// If upload schema columns + schema in warehouse columns < max columns , it is definitely a case of no exceeded columns.
+		if len(columnMap)+len(schemaInWarehouse[tableName]) < maxColumnCount {
+			continue
+		}
+
+		// Get distinct columns from schema in warehouse
+		newTableColumns := []string{}
```

Comment:
I would rename it to `additionalColumns` or something. Wdyt ? I was scratching my head for newTableColumns literally. So thought of sharing it with you.

Reply:
done. :P